### PR TITLE
take out osb prefixes through rote transformation of variable names and annotations

### LINKED ISSUE #834: Remove OSB prefixes
Based on a chat @pmorie and I had yesterday we should remove the "OSB" prefix on our properties since the chances of us supporting something else along side the current stuff seems pretty slim.

w.r.t. OSBGUID, we both liked `ExternalID`.

However, if based on https://github.com/kubernetes-incubator/service-catalog/issues/802 we use the OSB ID as our name then we may not need a OSBGUID property at all.

### PR DESCRIPTION
**Pull request by MHBauer**
check v1alpha1/types.go for the names and annotation changes

want to make sure the this gets tested in ci stuff

 pkg/apis/servicecatalog/v1alpha1/types.go is what you should look at. everythign else is based on that or generated code. specifically confirm the annotations changes.

Closes #834

### CODE REVIEW

**Comment by MHBauer** [comment]
anybody know where `MaxDBPerNode` comes from? It's not in the OSBAPI spec...

**Comment by pmorie** [comment]
@MHBauer I believe `MaxDBPerNode` was one of the conventional fields in metadata that used to be documented in the spec, but has since moved to the CF metadata conventions document.

**Comment by duglin** [comment]
wanna remove MaxDBsPerNode as part of this?

**Comment by pmorie** [comment]
> wanna remove MaxDBsPerNode as part of this?

I'd really prefer to just make this one transformations.  If we start accreting other changes into this, it'll be much harder to review.

**Comment by MHBauer** [comment]
I think this is done for now.

File: pkg/apis/servicecatalog/v1alpha1/types.go [hunk 1]
```diff
@@ -130,17 +130,17 @@ type ServiceClass struct {
...
+	MaxDBPerNode            *string  `json:"maxdbpernode,omitempty"`
+	DashboardOAuth2ClientID *string  `json:"dashboardOAuth2ClientID,omitempty"`
+	DashboardSecret         *string  `json:"dashboardSecret,omitempty"`
+	DashboardRedirectURI    *string  `json:"dashboardRedirectURI,omitempty"`
 	// Description is a short description of the service.
 	Description string `json:"description"`
 
 	// Metadata fields
-	OSBMetadata *runtime.RawExtension `json:"osbMetadata, omitempty"`
+	Metadata *runtime.RawExtension `json:"externalMetadata, omitempty"`
```
Comment (MHBauer) [inline pkg/apis/servicecatalog/v1alpha1/types.go]: `ServiceClass` already has a `metadata` field from the kubernetes `ObjectMeta`. I pseudocopied the `ExternalID` idea and added an `External` prefix. This was after I had renamed the field itself to `Metadata`.

File: pkg/apis/servicecatalog/types.go [hunk 2]
```diff
@@ -153,10 +153,10 @@ type ServicePlan struct {
 	ExternalID string
 
 	// OSB-specific
-	OSBFree bool
+	Free bool
```
Comment (duglin) [inline pkg/apis/servicecatalog/types.go]: What do people think about calling this `IsFree` ?

File: pkg/apis/servicecatalog/v1alpha1/types.go (hunk 1 above)
Comment (duglin) [inline pkg/apis/servicecatalog/v1alpha1/types.go]: +1 plus I'm not keen on the json name and golang name being so different.

**Comment by vaikas-google** [comment]
Has this been superceded by @pmorie PR: https://github.com/kubernetes-incubator/service-catalog/pull/846

**Comment by pmorie** [comment]
@vaikas-google nope, that one just removes a couple of fields, looks like morgan will have to rebase onto it

**Comment by pmorie** [comment]
@MHBauer LGTM modulo `ExternalMetadata`

**Comment by MHBauer** [comment]
Rebased. Should build and stuff. Codegen struggles.

**Comment by duglin** [comment]
LGTM if CI is happy but I still prefer `IsFree` to just `Free`

**Comment by pmorie** [comment]
> prefer IsFree to just Free

We have an unwritten convention not to use 'Is' as a prefix for boolean fields, which I opened a PR to document: https://github.com/kubernetes/community/pull/624

**Pull request merged**